nexus/src/db/pool.rs

```diff
@@ -69,6 +70,12 @@ impl Pool {
     }
 }
 
+const DISALLOW_FULL_TABLE_SCAN_SQL: &str =
+    "set disallow_full_table_scans = on; set large_full_scan_rows = 0;";
+#[cfg(test)]
+pub const ALLOW_FULL_TABLE_SCAN_SQL: &str =
```

Comment:
I assume this is because we want test code to be able to do full table scans to look for things that we might not want production indexes on.  Is that right?  That makes a lot of sense.  But I think we really *don't* want to allow full scans _from Nexus_ when running the test suite.  I don't think this will distinguish between those, will it?

Maybe this could be an optional config when you create the DataStore, and the test functions would use a different flag for this?

Reply:
I'm just adding a new constant called `ALLOW_FULL_TABLE_SCAN_SQL`, a test would have to explicitly grab this text and execute it as SQL themselves to use it, which I'm doing within a transaction. This way, **even within the same test**, all other operations still disallow full table scans.

Reply:
Oh, sorry, I misparsed this.  I thought we were defining the same constant differently based on the different configurations.  Thanks for clarifying.

I believe [these options last for the duration of the connection, not the current transaction](https://www.cockroachlabs.com/docs/stable/set-vars.html).  So when something grabs a connection, runs this, then puts the connection back in the pool, the modified options will apply to whoever gets that connection next.  While confirming that I discovered [you can do this on a per-transaction basis!](https://www.cockroachlabs.com/docs/stable/set-vars.html#set-local).

So is this doing the right thing already?

Reply:
Updated to `SET LOCAL` in 292ee96afd95db34af291af53581d8e90d9f96b2

Reply:
Sounds good.  I thought this might not work if you're not in a transaction when you use it, and that appears to be true.  All of the callers in this PR are inside a transaction when they use it.  So we're fine here.

It'd be neat if we had a more type-safe way to make sure this wasn't misused.  That seems outside the scope of this PR.  (I did suggest adding a doc comment above to mention this.)